CONTRIBUTING.md

```diff
@@ -1,248 +1,271 @@
+
+Run the following command:
+
+```bash
+cargo pgrx init
 ```
 
-#### 3.2. Clone and enter directory
+**MacOS Users**:
+
+- If you're on MacOS, you might need to configure Cargo for dynamic linking due to Postgres symbols not being available until runtime.
+
+  ```bash
+  mkdir -p ~/.cargo
+  echo '[target.'cfg(target_os="macos")']' >> ~/.cargo/config.toml
+  echo 'rustflags = ["-Clink-arg=-Wl,-undefined,dynamic_lookup"]' >> ~/.cargo/config.toml
+  ```
+
+### 2. Clone the `pg_vectorize` Repository
+
+Clone the repository and navigate into the project directory:
 
 ```bash
 git clone https://github.com/tembo-io/pg_vectorize.git
-
 cd pg_vectorize/extension
 ```
 
-#### 3.3. Install dependencies
+### 3. Set PostgreSQL Version
 
-From within the pg_vectorize/extension directory, run the following, which will install `pg_cron`, `pgmq`, and `pgvector`:
+Export the PostgreSQL version you are using (default is 17):
+
+```bash
+export PG_VERSION=17
+```
+
+### 4. Install Dependencies and Configure PostgreSQL
+
+Install the required PostgreSQL extensions and configure `postgresql.conf` by running:
 
 ```bash
 make setup
 ```
 
-#### 3.4. Compile and run `pg_vectorize`
+**This command will**:
+
+- Installs the necessary extensions: `pg_cron`, `pgvector`, `pgmq`, and `vectorscale`.
+- Updates `postgresql.conf` with the required settings.
+
+### 5. Build and Run `pg_vectorize`
+
+Compile the project and start PostgreSQL with the `pg_vectorize` extension:
 
 ```bash
 make run
 ```
 
-### 4. Confirm successful build
+This command will:
 
-#### 4.1. Check extension presence
+- Build the `pg_vectorize` extension.
+- Start PostgreSQL and bring you into the `psql` shell.
 
-Once the above command is run, you will be brought into Postgres via `psql`.
+### 6. Create the `vectorize` Extension
 
-Run the following command inside the `psql` console to enable the extensions:
+Inside the `psql` shell, create the `vectorize` extension:
 
 ```sql
-create extension vectorize cascade
+CREATE EXTENSION vectorize cascade;
 ```
 
-To list out the enabled extensions, run:
+## Verifying the Installation
+
+### 1. Check Installed Extensions
+
+List the installed extensions to verify that `vectorize` is properly installed:
 
 ```sql
 \dx
 ```
+
+Expected output:
+
 ```text
-                                      List of installed extensions
-    Name    | Version |   Schema   |                             Description
-------------+---------+------------+---------------------------------------------------------------------
+                           List of installed extensions
+   Name     | Version |   Schema   |               Description
+------------+---------+------------+------------------------------------------
  pg_cron    | 1.6     | pg_catalog | Job scheduler for PostgreSQL
- pgmq       | 1.1.1   | pgmq       | A lightweight message queue. Like AWS SQS and RSMQ but on Postgres.
+ pgmq       | 1.1.1   | pgmq       | A lightweight message queue.
```

Comment:
https://github.com/tembo-io/pgmq/blob/5eab98308047b477122219bf7df5ad710474f697/pgmq-extension/pgmq.control#L1

Reply:
If we're updating this section then let's update it to the latest version of PGMQ and use the most recent description.